### PR TITLE
block: add new consensusFormatValidation flag

### LINKED ISSUE #2135: Block: Add flag for consensus-specific format validation to Block Header
As pointed out by @fvictorio from Hardhat it would be good if there would be an option to enable/disable the [consensusFormatValidation()](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/block/src/header.ts#L305) check in the BlockHeader constructor.

By default this option should be set to `true` (so keep the behavior as we have now), I would be inclined to name as the method, so `consensusFormatValidation`.

### PR DESCRIPTION
**Pull request by acolytec3**
Fixes #2135

### CODE REVIEW

File: packages/block/src/header.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ import { CLIQUE_EXTRA_SEAL, CLIQUE_EXTRA_VANITY } from './clique'
 import type { BlockHeaderBuffer, BlockOptions, HeaderData, JsonHeader } from './types'
 import type { CliqueConfig } from '@ethereumjs/common'
 
+
```
Comment (gabrocheleau) [inline packages/block/src/header.ts:29]: Extra space

File: packages/block/src/types.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (gabrocheleau) [inline packages/block/src/types.ts:1]: Extra space

File: packages/block/src/types.ts [hunk 3]
```diff
@@ -69,6 +69,10 @@ export interface BlockOptions {
    * Will throw if provided on a non-PoA chain.
    */
   cliqueSigner?: Buffer
+  /**
+   * Perform consensus validation checks on header if set.  Defaults to true.
```
Comment (gabrocheleau) [inline packages/block/src/types.ts:73]:
```suggestion
   * Perform consensus validation checks on header if set. Defaults to true.
```

File: packages/block/test/header.spec.ts [hunk 4]
```diff
@@ -290,6 +290,66 @@ tape('[Block]: Header functions', function (t) {
...
+      () => BlockHeader.fromHeaderData({ stateRoot: badHash }),
+      (err: any) => err.message.includes('stateRoot must be 32 bytes'),
+      'throws on invalid state root hash length'
+    )
+    st.throws(
+      () => BlockHeader.fromHeaderData({ transactionsTrie: badHash }),
+      (err: any) => err.message.includes('transactionsTrie must be 32 bytes'),
+      'throws on invalid transactionsTrie root hash length'
+    )
+    let nonce = Buffer.alloc(5)
```
Comment (gabrocheleau) [inline packages/block/test/header.spec.ts:328]: Not a big deal, but I find the nonce reassignment a bit weird. Could we just either scope it differently or directly assign `fromHeaderData({nonce: Buffer.alloc(5)})` ?

**Review by gabrocheleau — APPROVED** [review]

**Review by gabrocheleau — APPROVED** [review]

**Comment by holgerd77** [comment]
Just a short additional LGTM from my side as well, @acolytec3 thanks for addressing! 🙂

**Pull request merged**